arlas-tagger/src/main/java/io/arlas/server/service/TagRefService.java

```diff
@@ -0,0 +1,149 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.arlas.server.service;
+
+import io.arlas.server.app.ArlasServerConfiguration;
+import io.arlas.server.exceptions.ArlasException;
+import io.arlas.server.exceptions.InvalidParameterException;
+import io.arlas.server.exceptions.NotAllowedException;
+import io.arlas.server.exceptions.NotFoundException;
+import io.arlas.server.kafka.TagKafkaProducer;
+import io.arlas.server.model.CollectionReference;
+import io.arlas.server.model.TagRefRequest;
+import io.arlas.server.model.TaggingStatus;
+import io.arlas.server.model.enumerations.AggregationTypeEnum;
+import io.arlas.server.model.enumerations.OperatorEnum;
+import io.arlas.server.model.request.*;
+import io.arlas.server.model.response.AggregationResponse;
+import io.arlas.server.model.response.UpdateResponse;
+import io.arlas.server.services.UpdateServices;
+import io.arlas.server.utils.ParamsParser;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.aggregations.bucket.MultiBucketsAggregation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+
+public class TagRefService extends KafkaConsumerRunner {
+    private Logger LOGGER = LoggerFactory.getLogger(TagRefService.class);
+    private final TagKafkaProducer tagKafkaProducer;
+    private final UpdateServices updateServices;
+    private TaggingStatus taggingStatus;
+
+    public TagRefService(ArlasServerConfiguration configuration, String topic, String consumerGroupId,
+                         TagKafkaProducer tagKafkaProducer, UpdateServices updateServices) {
+        super(configuration, topic, consumerGroupId);
+        this.tagKafkaProducer = tagKafkaProducer;
+        this.updateServices = updateServices;
+        this.taggingStatus = TaggingStatus.getInstance();
+    }
+
+    @Override
+    public void processRecords(ConsumerRecords<String, String> records) {
+        if (records.count() > 0) {
+            LOGGER.debug("Kafka polling returned " + records.count());
+        }
+
+        for (ConsumerRecord<String, String> record : records) {
+
+            try {
+                final TagRefRequest tagRequest = MAPPER.readValue(record.value(), TagRefRequest.class);
+                if (tagRequest.propagation == null) {
+                    LOGGER.info("No propagation requested: " + record.value());
+                    tagKafkaProducer.sendToExecuteTags(tagRequest);
+                } else {
+                    LOGGER.info("Propagation requested: " + record.value());
```

Comment:
anything that is printed at every message should be debug and not info

Reply:
Change committed.